source/backend/api/Helpers/Extensions/DispositionFileExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Linq;
+using System.Security.Claims;
+using Pims.Core.Extensions;
+using Pims.Dal.Entities;
+using Pims.Dal.Exceptions;
+using Pims.Dal.Repositories;
+
+namespace Pims.Api.Helpers.Extensions
+{
+    public static class DispositionFileExtensions
+    {
+        public static void ThrowMissingContractorInTeam(this PimsDispositionFile dispositionFile, ClaimsPrincipal principal, IUserRepository userRepository)
+        {
+            ArgumentNullException.ThrowIfNull(dispositionFile);
+
+            ArgumentNullException.ThrowIfNull(principal);
+
+            var pimsUser = userRepository.GetUserInfoByKeycloakUserId(principal.GetUserKey());
+
+            if (pimsUser?.IsContractor == true && !dispositionFile.PimsDispositionFileTeams.Any(x => x.PersonId == pimsUser.PersonId))
+            {
+                throw new UserOverrideException(UserOverrideCode.ContractorNotInTeam, "As a Contractor your user contact information should be assigned to the Disposition File's team");
```

Comment:
This doesn't seem to match the error text provided in the story:
As a contractor, you must add yourself as a team member to the file in order to create or save changes.

Also, the story specifies that this should be thrown as an error. The contract user should not be allowed to override this error - the system should completely disallow saving a file when a contractor is creating it and not on the team.

UserOverrideExceptions should only be used when the user is being shown a warning that can be optionally overriden.

Reply:
Updated